addons/web/static/src/views/form/form_controller.js

```diff
@@ -480,6 +480,9 @@ export class FormController extends Component {
     }
 
     async beforeUnload(ev) {
+        if (!this.model.config.resId){
+            return;
+        }
```

Comment:
You can't fix here. It will not save record when user close window for new records.